### PR TITLE
Add subscription option to set a default rule

### PR DESCRIPTION
**Pull request by norgitam**
Implements DefaultRuleDescription to allow setting a default rule for a subscription.

### CODE REVIEW

File: subscription_manager.go [hunk 1]
```diff
@@ -217,7 +224,9 @@ func (sm *SubscriptionManager) Put(ctx context.Context, name string, opts ...Sub
 		return nil, err
 	}
 
-	reqBytes = xmlDoc(reqBytes)
+	str := string(reqBytes)
+	str = strings.Replace(str, `xmlns:XMLSchema-instance="`+schemaInstance+`" XMLSchema-instance:type`, `xmlns:i="`+schemaInstance+`" i:type`, -1)
```
Comment (jhendrixMSFT) [inline subscription_manager.go:228]: Why is this necessary?

Comment (norgitam) [reply subscription_manager.go:228]: Service Bus will return a `"400, Details: The specified resource description is invalid"` if `DefaultRuleDescription` has a xmlns named `XMLSchema-instance` for some reason.

Same workaround is applied in the `putRule` function with this comment attached:

[TODO: fix the unmarshal / marshal of xml with this attribute or ask the service to fix it. This is sad, but works.](https://github.com/Azure/azure-service-bus-go/blob/d452ef94455eae1e3822ef47413cfcdbfa512e15/subscription_manager.go#L403)

**Comment by jhendrixMSFT** [comment]
Can you please add a test in `subscription_test.go` that uses this new configuration option?

**Comment by norgitam** [comment]
Added test.

Noticed that [testSubscriptionWithFalseRule](https://github.com/Azure/azure-service-bus-go/blob/dfc63648e72847d8db99808d311034f50c250355/subscription_test.go#L309) checks `FalseFilter` with `assert.Equal(t, "1=0", *rule.Filter.SQLExpression)` but FalseFilter is defined as `1!=1`. Not sure if it's intended or a bug, so just a heads up :)

File: subscription_manager.go [hunk 2]
```diff
@@ -553,6 +562,19 @@ func SubscriptionWithMessageTimeToLive(window *time.Duration) SubscriptionManage
 	}
 }
 
+// SubscriptionWithDefaultRuleDescription configures the subscription to set a
+// default rule
+func SubscriptionWithDefaultRuleDescription(filter FilterDescriber) SubscriptionManagementOption {
+	return func(s *SubscriptionDescription) error {
+		rule := &DefaultRuleDescription{
+			Filter: filter.ToFilterDescription(),
+			Name:   ptrString("rule0"),
```
Comment (jhendrixMSFT) [inline subscription_manager.go:571]: Is this name correct?  I did a little searching and all I can find is `$Default`.  Should that be used instead?

Comment (norgitam) [reply subscription_manager.go:571]: It was changed from `$Default` to `rule0` in [d452ef9](https://github.com/Azure/azure-service-bus-go/pull/172/commits/d452ef94455eae1e3822ef47413cfcdbfa512e15). For some reason when using the name `$Default` the filter is not set, so an empty filter is created. (`$` is actually an invalid character in an entity name, so it shouldn't work at all)

I can change the name to `_Default` or add a name field to `SubscriptionWithDefaultRuleDescription` so the caller can set an arbitrary name. Which option do you think would be most consistent/preferred?

Comment (jhendrixMSFT) [reply subscription_manager.go:571]: I see, thanks for clarifying.  Mostly I wasn't sure if the name can be arbitrary.  If it can be user-defined I like your second option of providing a way to configure it but falling back to some default (I do like `_Default` too).

Comment (norgitam) [reply subscription_manager.go]: Cool, I've added user-defined rule name to now. I think a fall back to `_Default` would just be misleading and prefer the user to get an error from service bus explaining what went wrong (incorrect character in name or whatever).
